src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -444,10 +444,10 @@ async Task<(List<ImportResource> Loaded, List<ImportResource> Conflicts)> Import
                 }
                 else if (importMode == ImportMode.IncrementalLoad)
                 {
-                    // dedup by last updated - keep all versions when version and lastUpdated exist on record.
+                    // dedup by last updated - take first version for single last updated, prefer large version.
                     var inputsDedupped = resources
-                        .GroupBy(_ => _.ResourceWrapper.ToResourceDateKey(_model.GetResourceTypeId, ignoreVersion: !_.KeepVersion || !_.KeepLastUpdated))
-                        .Select(_ => _.First())
+                        .GroupBy(_ => _.ResourceWrapper.ToResourceDateKey(_model.GetResourceTypeId, ignoreVersion: true))
+                        .Select(_ => _.OrderByDescending(_ => _.ResourceWrapper.Version).First())
```

Comment:
Is this a change of behavior? Will this no longer allow customers to have multiple versions with the same last updated?

Reply:
Yes. It is a fix for a bug. This only affects cases with incorrect input data.